src/dbnode/integration/write_quorum_test.go

```diff
@@ -184,6 +185,37 @@ func TestAddNodeQuorumOnlyOneNormalAndLeavingInitializingUp(t *testing.T) {
 	assert.Error(t, testWrite(topology.ConsistencyLevelAll))
 }
 
+func TestReplaceNodeWithShardsLeavingAndInitializingCountTowardsConsistencySet(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+
+	numShards := defaultNumShards
+	minShard := uint32(0)
+	maxShard := uint32(numShards - 1)
+
+	// nodes = m3db nodes
+	nodes, closeFn, testWrite := makeTestWrite(t, numShards, []services.ServiceInstance{
+		node(t, 0, newClusterShardsRange(minShard, maxShard, shard.Leaving)),
+		node(t, 1, newClusterShardsRange(minShard, maxShard, shard.Available)),
+		node(t, 2, newClusterShardsRange(minShard, maxShard, shard.Available)),
+		node(t, 3, newClusterShardsRange(minShard, maxShard, shard.Initializing)),
+	}, true)
+	defer closeFn()
+
+	require.NoError(t, nodes[0].StartServer())
+	defer func() { require.NoError(t, nodes[0].StopServer()) }()
+	require.NoError(t, nodes[1].StartServer())
+	defer func() { require.NoError(t, nodes[1].StopServer()) }()
+	require.NoError(t, nodes[3].StartServerDontWaitBootstrap())
+	defer func() { require.NoError(t, nodes[3].StopServer()) }()
+
+	// Writes succeed to one available node and on both leaving and initializing node.
```

Comment:
don't you need the negative case tested too? (i.e. writes fail in the conditions you're currently succeeding on if `isShardsLeavingAndInitializingCountTowardsConsistency` is set to false)

Reply:
I have added false flag in all other quorum test and they are behaving like they behaving initially.
I could have write -ve case but that would have been duplicate tests.